observability/otel/otel-collector/otel-collector-deployment-modes.adoc

```diff
@@ -0,0 +1,71 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="otel-collector-deployment-modes"]
+= Deployment Modes
+include::_attributes/common-attributes.adoc[]
+:context: otel-collector-deployment-modes
+
+toc::[]
+
+The OpenTelemetry Collector supports the following deployment modes: Deployment (default mode), DaemonSet, StatefulSet and Sidecar.
+
+== Configuring the Deployment Mode
+
+Deployment, DaemonSet and StatefulSet deploy the collector using the respective Kubernetes workload resource.
+The sidecar mode deploys the collector as a sidecar container to an existing pod.
+
+.OpenTelemetry Collector with DaemonSet deployment mode
+[source,yaml]
+----
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel
+spec:
+  mode: daemonset <1>
+----
+<1> The deployment mode. Supported values: `deployment` (default), `daemonset`, `statefulset` and `sidecar`
+
+== Sidecar Injection
+
+The collector can be injected into existing pod-based workloads by configuring the pod annotation `sidecar.opentelemetry.io/inject`.
+Injecting the collector as a sidecar allows for example accessing logfiles from inside the pod, using the xref:./otel-collector-receivers.adoc#filelog-receiver_otel-collector-receivers[filelog receiver].
+Another use case is to configure the application to send telemetry data to the sidecar over localhost (the sidecar runs in the same pod as the application), and the collector forwards the telemetry data to an external service via an encrypted and authenticated connection.
+
+Example:
+
+.OpenTelemetry Collector with sidecar deployment mode
+[source,yaml]
+----
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-otel-sidecar
+spec:
+  mode: sidecar <1>
+----
+<1> Configure sidecar deployment mode.
+
+.Annotate an existing pod to inject the collector sidecar
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-app
+  annotations:
+    sidecar.opentelemetry.io/inject: "true" <1>
+spec:
+  containers:
+  - ...
+----
+<1> Enable sidecar injection. Supported values: `true`, `false`, `otelcol-instance-name`, `namespace/otelcol-instance-name`
+
+The `sidecar.opentelemetry.io/inject` annotation supports the following values:
+
+*true*:: Inject the collector with the configuration of the OpenTelemetryCollector resource in the same namespace
+*my-otel-sidecar*:: Inject the collector with the configuration of the `my-otel-sidecar` OpenTelemetryCollector resource in the same namespace.
```

Comment:
`my-otel-sidecar` should be replaced by the actual instance name, not sure if this is clear or not